src/hazelweb/model/UndoHistory.re

```diff
@@ -464,7 +464,13 @@ let get_new_action_group =
       ~action: Action.t,
     )
     : option(UndoHistoryCore.action_group) =>
-  if (UndoHistoryCore.is_move_action(new_cursor_term_info)) {
+  /* they areparens and bsraces basically results with movements but
+    * it is an edit action, therefore there is a need to make sure that
+    * the action is not Construct(SCloseParens) or Construct(SCloseBraces)
+   */
+  if (UndoHistoryCore.is_move_action(new_cursor_term_info)
```

Comment:
can we make this a predicate on the action instead of having it compare the previous and new states?

Reply:
The entire if expression looks like it was redundant because the else branch was handling all possible cases. I just removed the if branch. Not sure why it was added, but I have removed it at the moment. It was added in dab69adceb8634f31e334ef922287795a66c0c03